object-storage-core/src/main/java/io/micronaut/objectstorage/response/UploadResponse.java

```diff
@@ -28,23 +29,37 @@
 @DefaultImplementation(DefaultUploadResponse.class)
 public interface UploadResponse<R> {
 
+    /**
+     * Creates an instance from the given parameters.
+     *
+     * @param key the key under which the file will be stored.
+     * @param eTag the entity tag of the object stored.
+     * @param nativeResponse upload response object.
+     * @param <R> Cloud vendor-specific upload response
+     *
+     * @return an upload response instance.
+     */
+    @NonNull
+    static <R> UploadResponse<R> of(@NonNull String key, @NonNull String eTag, @NonNull R nativeResponse) {
+        return new DefaultUploadResponse<>(key, eTag, nativeResponse);
+    }
```

Comment:
Is there a reason this was moved to the top of the interface? Better to have static methods at the bottom

Reply:
Basically for consistency with `UploadRequest` which has them all at the top.